components/script/dom/serviceworkercontainer.rs

```diff
@@ -53,55 +56,74 @@ impl ServiceWorkerContainerMethods for ServiceWorkerContainer {
         return self.controller.get()
     }
 
+    #[allow(unrooted_must_root)]
     // https://w3c.github.io/ServiceWorker/#service-worker-container-register-method
     fn Register(&self,
                 script_url: USVString,
-                options: &RegistrationOptions) -> Fallible<Root<ServiceWorkerRegistration>> {
+                options: &RegistrationOptions) -> Rc<Promise> {
+        let promise = Promise::new(self.global().r());
```

Comment:
@jdm I was chatting with @KiChjang on IRC and I was told not to use `.r()` anymore in my code. Would you want me to carry out the changes towards a non-`r()` code before you merge?